iac/instance.py

```diff
@@ -56,9 +63,12 @@ def from_aws_instance(inst: dict) -> InstanceSelf:
         return Instance(
             name=name,
             id=inst["InstanceId"],
+            state=inst["State"]["Name"],
             key_name=inst["KeyName"],
             tags=tags,
-            public_dns_name=inst["PublicDnsName"],
+            public_dns_name=inst.get("PublicDnsName", None),
```

Comment:
dns and ip may or may not be set on creation.... interestingly, they are not supposed to be set until state is "running", however, in practice, sometimes they are set. :100: